lib/active_triples/rdf_source.rb

```diff
@@ -107,6 +109,9 @@ def initialize(*args, &block)
       Array.wrap(self.class.type).each do |type|
         get_values(:type) << type unless get_values(:type).include?(type)
       end
+
+      # This needs to be set last to prevent infinite loops in construction
```

Comment:
If the callback is in place before line 107, then `get_values(:type) << ...` invokes `set`, which invokes `parent.property_will_change`, which eventually reaches `ActiveModel#attribute_will_change!` which, several calls later, leads to an invocation of https://github.com/mbarnett/active_fedora/blob/f4adcb9f3be4eaba20e489dd4f0010beb8a718cb/lib/active_fedora/fedora_attributes.rb#L46, which hasn't finished constructing, so `@relation` is as yet unset, so it calls the constructor again, and we do this until stack overflow.

I can flesh out that comment to better explain the situation. In general there's a chasing-your-own-tail trap in being notified of mutation in objects that haven't finished constructing.